packages/hathor-rpc-handler/src/rpcMethods/getBalance.ts

```diff
@@ -31,39 +38,47 @@ import { validateNetwork } from '../helpers';
  *
  * @throws {NotImplementedError} - If address indexes are specified, which is not implemented.
  * @throws {PromptRejectedError} - If the user rejects the balance confirmation prompt.
+ * @throws {InvalidParamsError} - If the request parameters are invalid.
  */
 export async function getBalance(
   rpcRequest: GetBalanceRpcRequest,
   wallet: HathorWallet,
   requestMetadata: RequestMetadata,
   promptHandler: TriggerHandler,
 ) {
-  const { network, tokens, addressIndexes } = rpcRequest.params;
+  try {
+    const params = getBalanceSchema.parse(rpcRequest.params);
```

Comment:
If you don't want to put the entire function inside a try catch and handle parsing errors at the end you could use [`safeParse`](https://zod.dev/?id=safeparse) which returns a success flag with the data or error.

What do you think?

Reply:
Way better, thanks! Done in 7b55ca64f1280c5e34fb2b337d8e54c0ce10c289